// client/index-browser.js

/* !!! This is code generated by Prisma. Do not edit directly. !!!
/* eslint-disable */
// biome-ignore-all lint: generated file

Object.defineProperty(exports, "__esModule", { value: true });

const {
  Decimal,
  objectEnumValues,
  makeStrictEnum,
  Public,
  getRuntime,
  skip
} = require('@prisma/client/runtime/index-browser.js')


const Prisma = {}

exports.Prisma = Prisma
exports.$Enums = {}

/**
 * Prisma Client JS version: 6.19.2
 * Query Engine version: c2990dca591cba766e3b7ef5d9e8a84796e47ab7
 */
Prisma.prismaVersion = {
  client: "6.19.2",
  engine: "c2990dca591cba766e3b7ef5d9e8a84796e47ab7"
}

Prisma.PrismaClientKnownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientKnownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)};
Prisma.PrismaClientUnknownRequestError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientUnknownRequestError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientRustPanicError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientRustPanicError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientInitializationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientInitializationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.PrismaClientValidationError = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`PrismaClientValidationError is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.Decimal = Decimal

/**
 * Re-export of sql-template-tag
 */
Prisma.sql = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`sqltag is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.empty = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`empty is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.join = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`join is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.raw = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`raw is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.validator = Public.validator

/**
* Extensions
*/
Prisma.getExtensionContext = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.getExtensionContext is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}
Prisma.defineExtension = () => {
  const runtimeName = getRuntime().prettyName;
  throw new Error(`Extensions.defineExtension is unable to run in this browser environment, or has been bundled for the browser (running in ${runtimeName}).
In case this error is unexpected for you, please report it in https://pris.ly/prisma-prisma-bug-report`,
)}

/**
 * Shorthand utilities for JSON filtering
 */
Prisma.DbNull = objectEnumValues.instances.DbNull
Prisma.JsonNull = objectEnumValues.instances.JsonNull
Prisma.AnyNull = objectEnumValues.instances.AnyNull

Prisma.NullTypes = {
  DbNull: objectEnumValues.classes.DbNull,
  JsonNull: objectEnumValues.classes.JsonNull,
  AnyNull: objectEnumValues.classes.AnyNull
}



/**
 * Enums
 */

exports.Prisma.TransactionIsolationLevel = makeStrictEnum({
  ReadUncommitted: 'ReadUncommitted',
  ReadCommitted: 'ReadCommitted',
  RepeatableRead: 'RepeatableRead',
  Serializable: 'Serializable'
});

exports.Prisma.AdminScalarFieldEnum = {
  id: 'id',
  email: 'email',
  passwordHash: 'passwordHash',
  name: 'name',
  lastLoginAt: 'lastLoginAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.UserScalarFieldEnum = {
  id: 'id',
  email: 'email',
  passwordHash: 'passwordHash',
  transactionPasswordHash: 'transactionPasswordHash',
  isActive: 'isActive',
  isBanned: 'isBanned',
  isVerified: 'isVerified',
  banReason: 'banReason',
  bannedAt: 'bannedAt',
  lastLoginAt: 'lastLoginAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.WalletScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  asset: 'asset',
  available: 'available',
  locked: 'locked',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SpotOrderScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  symbol: 'symbol',
  side: 'side',
  type: 'type',
  price: 'price',
  quantity: 'quantity',
  filledQuantity: 'filledQuantity',
  status: 'status',
  feeAsset: 'feeAsset',
  feeAmount: 'feeAmount',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TradeFillScalarFieldEnum = {
  id: 'id',
  orderId: 'orderId',
  userId: 'userId',
  symbol: 'symbol',
  price: 'price',
  quantity: 'quantity',
  feeAsset: 'feeAsset',
  feeAmount: 'feeAmount',
  side: 'side',
  createdAt: 'createdAt'
};

exports.Prisma.ContractPositionScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  symbol: 'symbol',
  side: 'side',
  entryPrice: 'entryPrice',
  amount: 'amount',
  duration: 'duration',
  profitability: 'profitability',
  expectedProfit: 'expectedProfit',
  expectedPayout: 'expectedPayout',
  status: 'status',
  exitPrice: 'exitPrice',
  actualProfit: 'actualProfit',
  result: 'result',
  createdAt: 'createdAt',
  expiresAt: 'expiresAt',
  closedAt: 'closedAt'
};

exports.Prisma.IEOProductScalarFieldEnum = {
  id: 'id',
  title: 'title',
  symbol: 'symbol',
  status: 'status',
  totalSupply: 'totalSupply',
  currentRaised: 'currentRaised',
  pricePerToken: 'pricePerToken',
  startDate: 'startDate',
  endDate: 'endDate',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.IEOInvestmentScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  productId: 'productId',
  amount: 'amount',
  tokens: 'tokens',
  status: 'status',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.ContractSessionControlScalarFieldEnum = {
  id: 'id',
  final: 'final',
  required: 'required',
  createdAt: 'createdAt'
};

exports.Prisma.MiningProductScalarFieldEnum = {
  id: 'id',
  hashRate: 'hashRate',
  currency: 'currency',
  averageDailyReturn: 'averageDailyReturn',
  minimumPurchase: 'minimumPurchase',
  maximumPurchase: 'maximumPurchase',
  duration: 'duration',
  status: 'status',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MiningInvestmentScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  productId: 'productId',
  amount: 'amount',
  hashRate: 'hashRate',
  dailyReturn: 'dailyReturn',
  totalReturn: 'totalReturn',
  startDate: 'startDate',
  endDate: 'endDate',
  status: 'status',
  lastPayoutDate: 'lastPayoutDate',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.WithdrawalRequestScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  asset: 'asset',
  chain: 'chain',
  address: 'address',
  amount: 'amount',
  fee: 'fee',
  arrival: 'arrival',
  status: 'status',
  txHash: 'txHash',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.TransferScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  asset: 'asset',
  amount: 'amount',
  fromAccount: 'fromAccount',
  toAccount: 'toAccount',
  createdAt: 'createdAt'
};

exports.Prisma.ExchangeTransactionScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  fromAsset: 'fromAsset',
  toAsset: 'toAsset',
  fromAmount: 'fromAmount',
  toAmount: 'toAmount',
  rate: 'rate',
  feeAsset: 'feeAsset',
  feeAmount: 'feeAmount',
  createdAt: 'createdAt'
};

exports.Prisma.ConversationScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  status: 'status',
  lastMessageAt: 'lastMessageAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.MessageScalarFieldEnum = {
  id: 'id',
  conversationId: 'conversationId',
  senderId: 'senderId',
  senderType: 'senderType',
  content: 'content',
  imageUrl: 'imageUrl',
  isRead: 'isRead',
  createdAt: 'createdAt'
};

exports.Prisma.VerificationRequestScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  name: 'name',
  idNumber: 'idNumber',
  frontIdUrl: 'frontIdUrl',
  backIdUrl: 'backIdUrl',
  status: 'status',
  rejectReason: 'rejectReason',
  reviewedBy: 'reviewedBy',
  reviewedAt: 'reviewedAt',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.DepositAddressScalarFieldEnum = {
  id: 'id',
  asset: 'asset',
  network: 'network',
  address: 'address',
  qrCodeUrl: 'qrCodeUrl',
  isActive: 'isActive',
  minAmount: 'minAmount',
  maxAmount: 'maxAmount',
  note: 'note',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.DepositRequestScalarFieldEnum = {
  id: 'id',
  userId: 'userId',
  asset: 'asset',
  network: 'network',
  amount: 'amount',
  certificateUrl: 'certificateUrl',
  status: 'status',
  rejectReason: 'rejectReason',
  reviewedBy: 'reviewedBy',
  reviewedAt: 'reviewedAt',
  completedAt: 'completedAt',
  txHash: 'txHash',
  createdAt: 'createdAt',
  updatedAt: 'updatedAt'
};

exports.Prisma.SortOrder = {
  asc: 'asc',
  desc: 'desc'
};

exports.Prisma.QueryMode = {
  default: 'default',
  insensitive: 'insensitive'
};

exports.Prisma.NullsOrder = {
  first: 'first',
  last: 'last'
};


exports.Prisma.ModelName = {
  Admin: 'Admin',
  User: 'User',
  Wallet: 'Wallet',
  SpotOrder: 'SpotOrder',
  TradeFill: 'TradeFill',
  ContractPosition: 'ContractPosition',
  IEOProduct: 'IEOProduct',
  IEOInvestment: 'IEOInvestment',
  ContractSessionControl: 'ContractSessionControl',
  MiningProduct: 'MiningProduct',
  MiningInvestment: 'MiningInvestment',
  WithdrawalRequest: 'WithdrawalRequest',
  Transfer: 'Transfer',
  ExchangeTransaction: 'ExchangeTransaction',
  Conversation: 'Conversation',
  Message: 'Message',
  VerificationRequest: 'VerificationRequest',
  DepositAddress: 'DepositAddress',
  DepositRequest: 'DepositRequest'
};

/**
 * This is a stub Prisma Client that will error at runtime if called.
 */
class PrismaClient {
  constructor() {
    return new Proxy(this, {
      get(target, prop) {
        let message
        const runtime = getRuntime()
        if (runtime.isEdge) {
          message = `PrismaClient is not configured to run in ${runtime.prettyName}. In order to run Prisma Client on edge runtime, either:
- Use Prisma Accelerate: https://pris.ly/d/accelerate
- Use Driver Adapters: https://pris.ly/d/driver-adapters
`;
        } else {
          message = 'PrismaClient is unable to run in this browser environment, or has been bundled for the browser (running in `' + runtime.prettyName + '`).'
        }

        message += `
If this is unexpected, please open an issue: https://pris.ly/prisma-prisma-bug-report`

        throw new Error(message)
      }
    })
  }
}

exports.PrismaClient = PrismaClient

Object.assign(exports, Prisma)
